src/yaml-languange-support/schema-utils.ts

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+/**
+ * An utility function for splitting text into lines.
+ */
+export function splitLines(text) {
+    return text.split(/\r\n?|\n/);
+} 
+
+/**
+ * A very easy utility function for decides whether the text is support to be a kubernetes yaml file, further
+ * implementations is needed to be more precise on the result rather than the regex test.
+ */
+export function isKubernetesYamlFile(text) {
+    return !!(/^kind\s*\:/g.exec(text) ||
+        /[^a-zA-Z]kind\s*\:/g.exec(text) ||
+        /^apiVersion\s*\:/g.exec(text) ||
+        /[^a-zA-Z]apiVersion\s*\:/g.exec(text)
+    );
+}
+
+/**
+ * An utility function for get apiVersion and kind from 'x-kubernetes-group-version-kind' node in kubernetes schema.
+ */
+export function parseApiVersionKind(groupKindNode) {
```

Comment:
I'm not sure what a `groupKindNode` is or where we get one from.  I take it this is an `any`, perhaps coming from a JSON or YAML load?  And we are extracting the interesting values into a well-typed structure?

Reply:
It is from kubernetes schema, see: https://github.com/garethr/openshift-json-schema/blob/master/v3.6.0/deployment.json#L2